frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -304,33 +361,122 @@ function formatPeriod(duration: bigint, freq: PeriodFrequency): string {
   return momentConverter(durationNumber, freqParam)
 }
 
-function formatCategoricalType(
-  x: number | bigint | StructRow,
-  field: Field
-): string {
-  // Serialization for pandas.Interval and pandas.Period is provided by Arrow extensions
+function formatPeriod(duration: number | bigint, field?: Field): string {
+  // Serialization for pandas.Period is provided by Arrow extensions
+  // https://github.com/pandas-dev/pandas/blob/70bb855cbbc75b52adcb127c84e0a35d2cd796a9/pandas/core/arrays/arrow/extension_types.py#L26
+  if (isNullOrUndefined(field)) {
+    logWarning("Field information is missing")
+    return String(duration)
+  }
+
+  const extensionName = field.metadata.get("ARROW:extension:name")
+  const extensionMetadata = field.metadata.get("ARROW:extension:metadata")
+
+  if (
+    isNullOrUndefined(extensionName) ||
+    isNullOrUndefined(extensionMetadata)
+  ) {
+    logWarning("Arrow extension metadata is missing")
+    return String(duration)
+  }
+
+  if (extensionName !== "pandas.period") {
+    logWarning(`Unsupported extension name for period type: ${extensionName}`)
+    return String(duration)
+  }
+
+  const parsedExtensionMetadata = JSON.parse(extensionMetadata as string)
+  const { freq } = parsedExtensionMetadata
+  return formatPeriodFromFreq(duration, freq)
+}
+
+/**
+ * Formats nested arrays and other objects to JSON.
+ */
+function formatObject(object: any, field?: Field): string {
+  if (field?.type instanceof Struct) {
+    // This type is used by python dictionary values
+
+    // Workaround: Arrow JS adds all properties from all cells
+    // as fields. When you convert to string, it will contain lots of fields with
+    // null values. To mitigate this, we filter out null values.
+
+    return JSON.stringify(object, (_key, value) => {
+      if (!notNullOrUndefined(value)) {
+        // Ignore null and undefined values ->
+        return undefined
+      }
+      if (typeof value === "bigint") {
+        return Number(value)
+      }
+      return value
+    })
+  }
+  return JSON.stringify(object, (_key, value) =>
+    typeof value === "bigint" ? Number(value) : value
```

Comment:
the only difference to the `stringify` above is that we don't check for `notNullOrUndefined`. Can't we just consolidate the cases, because I guess `value` here cannot be `nullOrUndefined` anyways?

Reply:
We probably can unify this, but I don't have a good grasp on what other data types might be handled via this function. I know that `dict` objects will have the type `Struct` -> and that's where we apply the ignore null/undefined logic since it will otherwise look broken (see the moved/added comment in the code).  

But for now, it might be safer to just add a todo comment here and look at unifying this in a later PR.